pkg/console/operator/operator.go

```diff
@@ -176,6 +190,20 @@ func NewConsoleOperator(
 		ToController("ConsoleOperator", recorder.WithComponentSuffix("console-operator"))
 }
 
+func isCapabilityEnabled(client configclientv1.ConfigV1Interface, expectedCap configv1.ClusterVersionCapability) bool {
+	cv, err := client.ClusterVersions().Get(context.TODO(), "version", metav1.GetOptions{})
```

Comment:
one-shot ClusterVersion fetch attempt here on operator initiation without a limiting `Context` has some risks.  A lower-stakes approach would be to not enable the OLM portions until we can confirm they're enabled, so you could have behavior like:

1. Console operator is coming up.
2. Console operator attempts to retrieve ClusterVersion with a 5m deadline, but this fails. 
3. Console operator assumes that OLM is not enabled, and goes on to operate its other aspects.
4. Follow-up attempt to check in on ClusterVersion succeeds, and it turns out that OLM is enabled.
5. Console operator starts operating on the OLM-gated stuff too.

You'll want something like that anyway, because admins may decide to enable capabilities post-install.  Only [post-install _disabling_ is not supported][1].

[1]: https://github.com/openshift/enhancements/blob/286e46ad8c1fc0baf9f940b21ea85381c3e54640/enhancements/installer/component-selection.md#capabilities-cannot-be-uninstalled

Reply:
I think @wking would just like this to be a bit more robust for future use. So in the event that a user re-enables OLM after the fact, so the pod will need to be restarted for the console to correctly see the resources.

I'm not entirely sure what to do about the informer since it's attached at startup, I updated the code a bit since we just needed to know if a resource exists for that informer. I also updated the error checking so that we don't always skip based off of initial state. If I understand this right, the informer won't be attached, but the lister will be queried after the fact, so if the user re-enables OLM, then the CSV query will go through.

Reply:
Right but in case the capability will get enabled post-installation we are not registering the informer. Please check my comment below.